anta/tests/bfd.py

```diff
@@ -285,3 +285,83 @@ def test(self) -> None:
         if up_failures:
             up_failures_str = "\n".join(up_failures)
             self.result.is_failure(f"\nFollowing BFD peers were down:\n{up_failures_str}")
+
+
+class VerifyBFDPeersRegProtocols(AntaTest):
+    """Verifies BFD peer(s) has the specified protocol(s) registered.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if IPv4 BFD peers are configured with specified protocol(s).
+    * Failure: The test will fail if IPv4 BFD peers are not found, BFD session is not established, specified protocol(s) are not registered on the BFD peer.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.bfd:
+      - VerifyBFDPeersRegProtocols:
+          bfd_peers:
+            - peer_address: 192.0.255.7
+              vrf: default
+              protocols:
+                - bgp
+    ```
+    """
+
+    name = "VerifyBFDPeersRegProtocols"
+    description = "Verifies IPv4 BFD peer(s) has the specified protocol(s) registered."
+    categories: ClassVar[list[str]] = ["bfd"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show bfd peers detail", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBFDPeersRegProtocols test."""
+
+        bfd_peers: list[BFDPeer]
+        """List of IPv4 BFD peers."""
+
+        class BFDPeer(BaseModel):
+            """Model for an IPv4 BFD peer."""
+
+            peer_address: IPv4Address
+            """IPv4 address of a BFD peer."""
+            vrf: str = "default"
+            """Optional VRF for BFD peer. If not provided, it defaults to `default`."""
+            protocols: list[Literal["bgp", "isis", "ospf", "eigrp"]]
+            """List of protocols to be verified."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyBFDPeersRegProtocols."""
+        # Initialize failure messages
+        failures: dict[Any, Any] = {}
+
+        # Iterating over BFD peers, extract the parameters and command output
+        for bfd_peer in self.inputs.bfd_peers:
+            peer = str(bfd_peer.peer_address)
+            vrf = bfd_peer.vrf
+            protocols = bfd_peer.protocols
+            bfd_output = get_value(
+                self.instance_commands[0].json_output,
+                f"vrfs..{vrf}..ipv4Neighbors..{peer}..peerStats..",
+                separator="..",
+            )
+
+            # Check if BFD peer configured
+            if not bfd_output:
+                failures[peer] = {vrf: "Not Configured"}
+                continue
+
+            # Check BFD peer status
+            if bfd_output.get("status") != "up":
+                failures[peer] = {vrf: {"status": bfd_output.get("status")}}
+                continue
```

Comment:
Let's remove this check and keep this test case only for registered protocols. We already check the status in the `VerifyBFDSpecificPeers` tests. Update the docstrings accordingly.

Reply:
Removed the check.